From 07ce8887dd8b5f2e21789404360ee9fd625ef5d3 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Mon, 23 Apr 2018 17:09:31 +0300
Subject: [PATCH 3622/5242] MLK-18121: ARM: dts: imx6qdl: Fix gpu
 power-domains ref

commit  91c18a66bc84fa558be90e9cc7fae215f57bfb01 from
https://source.codeaurora.org/external/imx/linux-imx.git

Upstream refactored GPC power domains to have a nested pgc node, this
means that the pu power domain needs to be referenced differently.

This fixes warnings like these:

OF: /soc/gpu@00130000: could not get #power-domain-cells for
/soc/aips-bus@02000000/gpc@020dc000

This warning probably also means that GPU power management is not done
correctly.

Old bindings are still used on imx6s*

Fixes: 0a9a9a732f88 ("MGS-3705-1 gpu: dts: enable gpu devices for imx6")
Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Reviewed-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dl.dtsi |    2 +-
 arch/arm/boot/dts/imx6q.dtsi  |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl.dtsi b/arch/arm/boot/dts/imx6dl.dtsi
index 6bedf8a..08c2acb 100644
--- a/arch/arm/boot/dts/imx6dl.dtsi
+++ b/arch/arm/boot/dts/imx6dl.dtsi
@@ -105,7 +105,7 @@
 				      "gpu3d_shader_clk";
 			resets = <&src 0>, <&src 3>;
 			reset-names = "gpu3d", "gpu2d";
-			power-domains = <&gpc 1>;
+			power-domains = <&pd_pu>;
 		};
 
 		ocram: sram@905000 {
diff --git a/arch/arm/boot/dts/imx6q.dtsi b/arch/arm/boot/dts/imx6q.dtsi
index 1a3d4a6..9b66420 100644
--- a/arch/arm/boot/dts/imx6q.dtsi
+++ b/arch/arm/boot/dts/imx6q.dtsi
@@ -124,7 +124,7 @@
 				      "gpu3d_clk", "gpu3d_shader_clk";
 			resets = <&src 0>, <&src 3>, <&src 3>;
 			reset-names = "gpu3d", "gpu2d", "gpuvg";
-			power-domains = <&gpc 1>;
+			power-domains = <&pd_pu>;
 		};
 
 		ocram: sram@905000 {
-- 
1.7.9.5

